### DISCUSSION TITLE
TypeScript error __loadedType can not be named

### DISCUSSION THREAD

**Discussion by Kalos-S** [mikro-orm/mikro-orm#6060] [General]
### Describe the bug

With the TypeScript compiler option 'declaration' set to true returning a reference to any loaded entity generated with the ref() helper in a public method of an exported class results in this TypeScript error:
Return type of public method from exported class has or is using name '__loadedType' from external module "[...]/repro/node_modules/@mikro-orm/core/typings" but cannot be named.

### Reproduction

// tsconfig.json
{
  "compilerOptions": {
    "declaration": true
  }
}

// test.ts
import { type Loaded, ref } from "@mikro-orm/core";

export class Test {
  async test() {
    // Entity definition does not matter
    const loaded = {} as Loaded<{}>;

    return ref(loaded);
  }
}

### What driver are you using?

None

### MikroORM version

6.3.10

### Node.js version

20.16.0

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
You need to use explicit return type in those places.
